src/components/List_Products.js

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from 'react';
+import { saveItem } from '../lib/api';
+import { db } from '../lib/firebase';
+import { collection, onSnapshot } from 'firebase/firestore';
+
+function ListProducts() {
+  const [items, setItems] = useState(null);
+  const [itemName, setItemName] = useState('');
+
+  /* Get items */
+  useEffect(() => {
+    const unsubscribe = onSnapshot(collection(db, 'Products'), (snapshot) => {
+      let products = [];
+
+      snapshot.forEach((doc) => {
+        products.push({ ...doc.data(), id: doc.id });
+      });
+
+      console.log(products);
+      setItems(products);
+
+      return () => {
```

Comment:
I'm a bit lost to what this `return` function is doing. is it supposed to be a clean up function for the `useEffect`? if yes, it should go outside the `unsubscribe` variable. However, I feel like at this point the return function is not making a difference. Lmk what y'all think

Reply:
I agree with you, Camila. I believe this is not a correct place where you should "unsubscribe".  I had some errors: _Can't perform a React state update on an unmounted component_

![image](https://user-images.githubusercontent.com/8689897/149642232-fc8ae1df-2c89-45bd-aead-8656da966c18.png)

Reply:
An possible suggestion:

```js
  useEffect(() => {
    const unsubscribe = onSnapshot(collection(db, 'Products'), (snapshot) => {
      //... more code
    });
    return () => {
      unsubscribe();
    };
  }, []);
```